### PR TITLE
Refactored metadata builder

### PR DESCRIPTION
**Pull request by Henry-bee**

### CODE REVIEW

File: rust/src/builders/tx_builder.rs [hunk 1]
```diff
@@ -712,6 +717,25 @@ impl TransactionBuilder {
         Ok(())
     }
```
Comment (Henry-bee) [inline rust/src/builders/tx_builder.rs:719]: Or should we simply remove them?

File: rust/src/builders/metadata_builder.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+#[wasm_bindgen]
+pub struct TxMetadataBuilder {
+    metadata: GeneralTransactionMetadata,
+}
+
+#[wasm_bindgen]
+impl TxMetadataBuilder {
+    pub fn new() -> Self {
```
Comment (SebastienGllmt) [inline rust/src/builders/metadata_builder.rs:15]: Probably it's better to have two constructors:
1. `pub fn empty()` that works like what you have now
2. `pub fn from(metadata: &GeneralTransactionMetadata)`

File: rust/src/builders/tx_builder.rs (hunk 1 above)
Comment (SebastienGllmt) [reply rust/src/builders/tx_builder.rs:719]: I would just remove them

File: rust/src/builders/metadata_builder.rs (hunk 2 above)
Comment (Henry-bee) [reply rust/src/builders/metadata_builder.rs:15]: That's a great suggestion. Let me add that

**Pull request closed without merging**